packages/wallets/particle/src/adapter.ts

```diff
@@ -0,0 +1,179 @@
+import {
+    BaseMessageSignerWalletAdapter,
+    WalletAccountError,
+    WalletConfigError,
+    WalletConnectionError,
+    WalletDisconnectionError,
+    WalletLoadError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignMessageError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import { SolanaWallet } from '@particle-network/solana-wallet';
```

Comment:
You can't import this here except as types. All third party libraries must be loaded async.